scripts/start-terra-site/site.webpack.config.js

```diff
@@ -0,0 +1,42 @@
+const path = require('path');
+const fs = require('fs');
+const webpack = require('webpack');
+// const generateComponentConfig = require('../generate-component-config/generate-component-configDyanmic');
+
+const isFile = filePath => (fs.existsSync(filePath) && !fs.lstatSync(filePath).isDirectory());
+
+/* Get the site configuration */
+const siteConfigPath = path.resolve(path.join(process.cwd(), 'site.config.js'));
+// eslint-disable-next-line import/no-dynamic-require
+const siteConfig = isFile(siteConfigPath) ? require(siteConfigPath) : require('../../src/config/site.config');
+
+/* Get the webpacak configuration */
+let webpackConfigPath;
+if (siteConfig.webpackConfig) {
+  webpackConfigPath = path.resolve(path.join(process.cwd(), `${siteConfig.webpackConfig}`));
+}
+// eslint-disable-next-line import/no-dynamic-require
+const webpackConfig = isFile(webpackConfigPath) ? require(webpackConfigPath) : require('../../src/config/webpack.config');
+
+/* Get the component configuration */
+let componentConfigPath = path.resolve(path.join(process.cwd(), siteConfig.componentConfig));
+// let componentConfig;
+if (!isFile(componentConfigPath)) {
+  componentConfigPath = undefined;
+}
+// else {
```

Comment:
I want to build the generatedComponentConfig from the information given in the site.config file, however I haven't figured out a clean way to import these components. I locally have the changes to generate the correct config, but the components result in a string vs a function. I tried to dynamically import the  components during the generatedComponentConfig, however one needs to use `require` syntax which ultimately leads to the need for pre-compiled examples (I think). The better option would be to load these components within the Index page or RoutedSite component, but I ran into async errors. I got around this with importing the component config file by saving the component to state, but this does not seem like a good option for multiple dynamic imports.